src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -2254,29 +2383,54 @@ static bool isMixableArgument(BoundExpression argument)
                 return true;
             }
 
-            static EscapeArgument getReceiver(MethodSymbol? method, BoundExpression receiver)
+            static EscapeArgument getReceiver(in MethodInfo methodInfo, BoundExpression receiver)
             {
-                if (method is FunctionPointerMethodSymbol)
+                // When there is compound usage the receiver is used once but both the get and 
+                // set methods are invoked. Return the most permissive of the two in order to get
+                // the complete set of ref safety errors.
```

Comment:
I would prefer to change "return the most permissive" to something like "prefer a this parameter which is a writable reference, if any". The reason is that I don't find "most permissive" to be a very precise concept in the context of ref safety. At a glance I thought we were going to compare the ScopedKind or something of each this parameter.